cromwell-drs-localizer/src/main/scala/drs/localizer/DrsLocalizerMain.scala

```diff
@@ -50,89 +34,36 @@ object DrsLocalizerMain extends IOApp {
-    val marthaErrorMsg = s"Something went wrong while trying to resolve $drsUrl through Martha $marthaUrl."
-
-    logger.info(s"Resolving DRS uri `$drsUrl` through Martha")
+class DrsLocalizerMain(drsUrl: String, downloadLoc: String, requesterPaysId: Option[String]) {
 
-    val scopedCredentials = GoogleCredentials.getApplicationDefault().createScoped(UserInfoScopes.asJava)
-    val accessToken = scopedCredentials.refreshAccessToken().getTokenValue
+  private final val RequesterPaysErrorMsg = "Bucket is requester pays bucket but no user project provided."
+  private final val ExtractGcsUrlErrorMsg = "No resolved url starting with 'gs://' found from Martha response!"
 
-    val request = sttp
-      .headers(Map(HeaderNames.ContentType -> MediaTypes.Json, "Authorization" -> s"Bearer $accessToken"))
-      .body(requestBody)
-      .post(marthaUrl)
-      .response(asJson[MarthaResponse])
-      .readTimeout(5.minutes)
+  final val requestTemplate = """{"url": "${drsPath}"}"""
 
-    val response = request.send()
+  val httpClientBuilder: HttpClientBuilder = HttpClientBuilder.create()
 
-    response.body match {
-      // non-2xx status code
-      case Left(error) => IO.raiseError(new Exception(s"$marthaErrorMsg Expected 200 but got ${response.code} from Martha. ${samErrorResponseMsg(error)}"))
-      case Right(marthaResponseEither) => {
-        logger.info("Received successful response from Martha")
-        marthaResponseEither match {
-          case Left(deserializationError) => IO.raiseError(new Exception(s"$marthaErrorMsg Failed to parse Martha response. " +
-            s"Deserialization error: ${deserializationError.message}"))
-          case Right(marthaResponse) => IO(marthaResponse)
-        }
-      }
+  def getDrsPathResolver: IO[LocalizerDrsPathResolver] = {
+    IO.pure {
+      val marthaUrl = sys.env("MARTHA_URL")
+      val drsConfig = DrsConfig(marthaUrl, requestTemplate)
+      new LocalizerDrsPathResolver(drsConfig, httpClientBuilder)
     }
   }
 
 
-  private def extractFirstGcsUrl(urlArray: Array[Url]): IO[String] = {
-    val urlOption = urlArray.find(_.url.startsWith(GcsScheme))
-
-    urlOption match {
-      case Some(url) => IO(url.url)
-      case None => IO.raiseError(new Exception(ExtractGcsUrlErrorMsg))
-    }
-  }
-
+  /*
+     Bash to download the GCS file using `gsutil`
+   */
+  def downloadScript(gcsUrl: String,
+                     downloadLoc: String,
+                     saJsonPathOption: Option[Path],
+                     requesterPaysProjectIdOption: Option[String]): String = {
 
-  private def downloadFileFromGcs(gcsUrl: String,
-                                  serviceAccountJsonOption: Option[String],
-                                  downloadLoc: String,
-                                  requesterPaysProjectIdOption: Option[String]) : IO[ExitCode] = {
+    def gcsCopyCommand(flag: String = ""): String = s"gsutil $flag cp $gcsUrl $downloadLoc"
```

Comment:
These are inputs from inside Cromwell (see [PipelinesParameterConversions.scala](https://github.com/broadinstitute/cromwell/blob/d380361da9cdc89ac35525c8a7d6bd4ae1227871/supportedBackends/google/pipelines/v2alpha1/src/main/scala/cromwell/backend/google/pipelines/v2alpha1/PipelinesParameterConversions.scala#L31-L47)) not users.

Reply:
Everything is passed from within Cromwell but input parameters ultimately originate from users. Since there's probably upstream vetting and we're running this under a user's profile I don't think it's a security problem to leave this as-is, just some small scope for surprises later on if metacharacters do sneak in  🤷‍♂️